### DISCUSSION TITLE
GraalVM adding TypeHint for private packages

### DISCUSSION THREAD

**Discussion by ThraaxSession** [micronaut-projects/micronaut-core#8111] [Q&A] [answered]
### Issue description

So my app is almost working as native image. 
Now I've got the issue that I can't add a `@TypeHint` for a private class.
Is there anyway I can add it anyway?

```
03.10.2022 11:25:23.802 [main] ERROR t.iatlas.spaceup.services.SshService - **java.lang.ClassNotFoundException: com.jcraft.jsch.DHGEX256**
03.10.2022 11:25:23.802 [main] ERROR t.iatlas.spaceup.services.SshService - SSH Session is down. Will try to reconnect.
03.10.2022 11:25:23.821 [main] INFO  t.iatlas.spaceup.services.SshService - Authenticate SSH via password!
03.10.2022 11:25:23.868 [main] ERROR t.iatlas.spaceup.services.SshService - **java.lang.ClassNotFoundException: com.jcraft.jsch.DHGEX256**
03.10.2022 11:25:23.868 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: session is down
com.jcraft.jsch.JSchException: session is down
```

**Comment by graemerocher** [comment, marked as answer]
Use `typeNames` member https://docs.micronaut.io/latest/api/io/micronaut/core/annotation/TypeHint.html#typeNames--

**Reply by ThraaxSession** [reply]
@graemerocher I add the class as Typename? Or the full package?
```
@TypeHint(
    value = [
        ...
    ],
    accessType = [
        TypeHint.AccessType.ALL_DECLARED_CONSTRUCTORS,
        TypeHint.AccessType.ALL_PUBLIC,
        TypeHint.AccessType.ALL_DECLARED_FIELDS,
        TypeHint.AccessType.ALL_DECLARED_METHODS,
    ],
    typeNames = ["DHGEX256"]
)
```

**Reply by graemerocher** [reply]
Fully qualified name

**Reply by ThraaxSession** [reply]
Awesome! Thank you! Now I have to add x other ciphers. :D

**Reply by ThraaxSession** [reply]
@graemerocher might it be possible to add a wildcard for the fully qualified name? like "com.jcraft.jsch.*"? Is it technical possible? It would help a lot.

**Reply by graemerocher** [reply]
Unfortunately I don't believe this is possible as it would require classpath scanning today

**Reply by ThraaxSession** [reply]
I ask because my application starts now, but I don't get any result via JSCH. So I guess, something is missing, but it doesn't blame me. Any Idea how I can figure it out?

**Reply by graemerocher** [reply]
Try run with the agent enabled `./gradlew run -Pagent` 

See https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#agent-support
